alert-common/src/main/java/com/synopsys/integration/alert/common/email/MessageContentGroupCsvCreator.java

```diff
@@ -0,0 +1,178 @@
+/**
+ * alert-common
+ *
+ * Copyright (c) 2019 Synopsys, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.synopsys.integration.alert.common.email;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Component;
+
+import com.synopsys.integration.alert.common.message.model.ComponentItem;
+import com.synopsys.integration.alert.common.message.model.LinkableItem;
+import com.synopsys.integration.alert.common.message.model.MessageContentGroup;
+import com.synopsys.integration.alert.common.message.model.ProviderMessageContent;
+
+@Component
+public class MessageContentGroupCsvCreator {
+    private static final String UNDEFINED_VALUE = "N/A";
+    private static final String MULTI_VALUE_DELIMITER = " / ";
+
+    public String createCsvString(MessageContentGroup messageContentGroup) {
+        LinkableItem commonProvider = messageContentGroup.getCommonProvider();
+        LinkableItem commonTopic = messageContentGroup.getCommonTopic();
+        List<ProviderMessageContent> contents = messageContentGroup.getSubContent();
+
+        StringBuilder csvBuilder = new StringBuilder();
+        List<String> columnNames = createColumnNames(commonProvider, commonTopic, contents);
+        appendLine(csvBuilder, columnNames);
+
+        List<List<String>> rowValues = createRowValues(commonProvider, commonTopic, contents);
+        for (List<String> row : rowValues) {
+            appendLine(csvBuilder, row);
+        }
+
+        return csvBuilder.toString();
+    }
+
+    // Provider | Topic Name | Sub Topic Name | Operation | Component Name | Sub Component Name | Component URL | Category Name | Category Grouping Attribute Name | Item URL
```

Comment:
Leaving this here as a guide for anyone who wants to have an idea of what the output looks like.